src/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -226,6 +226,16 @@ public function __construct(EntityManagerInterface $em, ClassMetadata $class)
         );
     }
 
+    protected function isActualFilterHash(): bool
```

Comment:
Neither methods are ever overridden and should therefore be `final`.

Regarding the naming of this one, I think this might be closer to what you actually mean:

```suggestion
    protected function isFilterHashUpToDate(): bool
```

Reply:
fixed

Reply:
Both methods should be `final`.

Reply:
> Both methods should be `final`.

corrected